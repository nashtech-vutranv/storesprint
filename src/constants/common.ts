export const ERROR_FIELD_INVALID = 'FIELD_INVALID'
export const URL_REGEX = new RegExp(
  '(^https?:\\/\\/)((www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b([-a-zA-Z0-9()@:%_\\+.~#?&//=]*))',
  'g'
)
export const extractDomainFromUrl = (url: string) => {
  if (url.match(URL_REGEX)) {
    const match = URL_REGEX.exec(url)
    return match ? match[2].toLowerCase() : ''
  }
  return ''
}

export const FORMAT_DATE = 'DD-MM-yyyy HH:mm'

export const FORMATE_DATE_NO_HOURS = 'DD/MM/yyyy'
