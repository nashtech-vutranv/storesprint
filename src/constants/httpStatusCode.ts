export const INTERNAL_SERVER_ERROR = 500
export const NOT_IMPLEMENTED_ERROR = 501
export const BAD_GATEWAY_ERROR = 502
export const SERVICE_UNAVAILABLE_ERROR = 503
export const GATEWAY_TIMEOUT_ERROR = 504
export const HTTP_VERSION_NOT_SUPPORTED_ERROR = 505
export const BAD_REQUEST_ERROR = 400
export const UNAUTHORIZED_ERROR = 401
export const FORBIDDEN_ERROR = 403
export const NOT_FOUND_ERROR = 404
export const CONFLICT_ERROR = 409