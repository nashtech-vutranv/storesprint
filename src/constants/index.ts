export * from './common'
export * from './layout'
export * from './order'
export * from './integrateMarketplace'
export * from './language'
export * from './pagination'
export * from './seo-url'
export * from './tabItem'
export * from './topbar'
export * from './menu'
export * from './routes'
export * from './permissions'
export * from './httpStatusCode'