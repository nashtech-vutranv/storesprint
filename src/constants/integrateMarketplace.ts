export const MARKETPLACE_INTEGRATION_KEY = 'marketplaceIntegration'
export const MARKETPLACE_INTEGRATION_STATE_KEY = 'integrationState'