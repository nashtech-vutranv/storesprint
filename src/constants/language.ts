import {LanguageKey, LanguageValue, ILanguageSample} from '../interface/language' 

export const listLanguageSelected: ILanguageSample[] = [
  {id: 1, value: LanguageValue.english, key: LanguageKey.english},
  {id: 2, value: LanguageValue.french, key: LanguageKey.french}
]