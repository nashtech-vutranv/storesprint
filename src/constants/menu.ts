import {LabelValue, MenuItemType} from '../interface'
import {
  ROUTE_ORG,
  ROUTE_PRODUCT,
  ROUTE_WAREHOUSE,
  ROUTE_MARKETPLACE,
  ROUTE_MARKETPLACE_RELATIONSHIP,
  ROUTE_ASSIGNED_PRODUCTS_TO_MARKETPLACES,
  ROUTE_LISTING_STATUS,
  ROUTE_PRODUCT_PROPERTIES,
  ROUTE_DEFAULT_PROPERTY_VALUE,
  ROUTE_PRODUCT_CATEGORY_MAPPINGS,
  ROUTE_COURIER_MAPPINGS,
  ROUTE_PRODUCT_PROPERTY_MAPPINGS,
  ROUTE_ORDER,
  ROUTE_DELIVERY_TYPE_MAPPINGS,
  ROUTE_UPS,
} from './routes'
import {PERMISSIONS} from './permissions'

const LABELS: LabelValue = {
  management: 'management_title',
  organizations: 'organization_title',
  warehouse: 'navigator_title_warehouse',
  products: 'navigator_title_products',
  productProperties: 'navigator_title_products_properties',
  configuration: 'navigator_title_configuration',
  // users: 'navigator_title_users',
  // aggregators: 'navigator_title_aggregators',
  marketplaces: 'navigator_title_marketplaces',
  // relationshipAM: 'navigator_title_relationshipAM',
  marketplaceRelationships: 'navigator_title_marketplace_relationships',
  assignedProducts: 'navigator_title_products_marketplaces',
  orders: 'navigator_title_orders',
  listingStatus: 'navigator_title_listing_status',
  defaultPropertyValue: 'navigator_title_default_property_value',
  productCategoryMappings: 'navigator_title_product_category_mappings',
  courierMappings: 'navigate_title_courier_mappings',
  marketplaceProductCategories: 'navigate_title_marketplace_product_categories',
  returnsRefunds: 'navigator_title_returns_refunds',
  deliveryTypeMappings: 'navigator_title_delivery_type_mappings',
}

export const MENU_ITEMS: MenuItemType[] = [
  {
    key: 'apps-management',
    label: LABELS.management,
    isTitle: false,
    icon: 'uil-suitcase',
    permissions: [
      PERMISSIONS.view_organization_list,
      PERMISSIONS.view_product_property_list,
      PERMISSIONS.view_warehouse_list,
      PERMISSIONS.view_product_list,
      PERMISSIONS.view_default_property_value_list,
      PERMISSIONS.view_marketplace_list,
      PERMISSIONS.view_marketplace_relationship_list,
      PERMISSIONS.view_product_category_mapping_list,
      PERMISSIONS.view_product_property_mapping_list,
      PERMISSIONS.view_delivery_mapping_list,
      PERMISSIONS.view_courier_mapping_list,
      PERMISSIONS.view_assign_products_to_marketplaces,
      PERMISSIONS.view_listing_status,
      PERMISSIONS.view_order_list,
    ],
    children: [
      {
        key: 'apps-organizations',
        label: LABELS.organizations,
        url: ROUTE_ORG.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_organization_list],
      },
      {
        key: 'apps-product-properties',
        label: LABELS.productProperties,
        url: ROUTE_PRODUCT_PROPERTIES.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_product_property_list],
      },
      {
        key: 'apps-warehouse',
        label: LABELS.warehouse,
        url: ROUTE_WAREHOUSE.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_warehouse_list],
      },
      {
        key: 'apps-products',
        label: LABELS.products,
        url: ROUTE_PRODUCT.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_product_list],
      },
      {
        key: 'apps-default-property-values',
        label: LABELS.defaultPropertyValue,
        url: ROUTE_DEFAULT_PROPERTY_VALUE.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_default_property_value_list],
      },
      // {
      //   key: 'apps-aggregators',
      //   label: LABELS.aggregators,
      //   url: ROUTE_AGGREGATOR.ROOT,
      //   parentKey: 'apps-management',
      // },
      {
        key: 'apps-marketplaces',
        label: LABELS.marketplaces,
        url: ROUTE_MARKETPLACE.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_marketplace_list],
      },
      // {
      //   key: 'apps-aggregator-marketplace',
      //   label: LABELS.relationshipAM,
      //   url: ROUTE_RELATIONSHIP_AM.ROOT,
      // },
      {
        key: 'apps-marketplace-relationships',
        label: LABELS.marketplaceRelationships,
        url: ROUTE_MARKETPLACE_RELATIONSHIP.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_marketplace_relationship_list],
      },
      {
        key: 'apps-product-category-mappings',
        label: LABELS.productCategoryMappings,
        url: ROUTE_PRODUCT_CATEGORY_MAPPINGS.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_product_category_mapping_list],
      },
      {
        key: 'apps-marketplace-product-categories',
        label: LABELS.marketplaceProductCategories,
        url: ROUTE_PRODUCT_PROPERTY_MAPPINGS.ROOT_NO_PARAMS,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_product_category_mapping_list],
      },
      {
        key: 'apps-delivery-type-mappings',
        label: LABELS.deliveryTypeMappings,
        url: ROUTE_DELIVERY_TYPE_MAPPINGS.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_delivery_mapping_list],
      },
      {
        key: 'apps-courier-mappings',
        label: LABELS.courierMappings,
        url: ROUTE_COURIER_MAPPINGS.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_courier_mapping_list],
      },
      {
        key: 'apps-assigned-products',
        label: LABELS.assignedProducts,
        url: ROUTE_ASSIGNED_PRODUCTS_TO_MARKETPLACES.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_assign_products_to_marketplaces],
      },
      {
        key: 'apps-listing-status',
        label: LABELS.listingStatus,
        url: ROUTE_LISTING_STATUS.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_listing_status],
      },
      {
        key: 'apps-orders',
        label: LABELS.orders,
        url: ROUTE_ORDER.ROOT,
        parentKey: 'apps-management',
        permissions: [PERMISSIONS.view_order_list],
      },
      // {
      //   key: 'apps-returnrefund',
      //   label: LABELS.returnsRefunds,
      //   url: '/returnrefund',
      //   parentKey: 'apps-management',
      // },
    ],
  },
  // {
  //   key: 'apps-configuration',
  //   label: LABELS.configuration,
  //   isTitle: false,
  //   icon: 'dripicons-gear',
  //   children: [
  //     {
  //       key: 'app-users',
  //       label: LABELS.users,
  //       url: ROUTE_USER.ROOT,
  //       parentKey: 'apps-configuration',
  //     },
  //   ],
  // },
]

export const UPS_MENU_ITEMS: MenuItemType[] = [
  {
    key: 'ups-application-setup',
    label: 'Application Setup',
    isTitle: false,
    icon: 'dripicons-gear',
    permissions: [
      PERMISSIONS.ups_view_application_list,
      PERMISSIONS.ups_view_module_list,
      PERMISSIONS.ups_view_role_list,
      PERMISSIONS.ups_view_permission_list,
    ],
    children: [
      {
        key: 'ups-applications',
        label: 'Applications',
        url: ROUTE_UPS.APPLICATION_SETUP.APPLICATION,
        parentKey: 'ups-application-setup',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_application_list],
      },
      {
        key: 'ups-modules',
        label: 'Modules',
        url: ROUTE_UPS.APPLICATION_SETUP.MODULES,
        parentKey: 'ups-application-setup',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_module_list],
      },
      {
        key: 'ups-roles',
        label: 'Roles',
        url: ROUTE_UPS.APPLICATION_SETUP.ROLES,
        parentKey: 'ups-application-setup',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_role_list],
      },
      {
        key: 'ups-permissions',
        label: 'Permissions',
        url: ROUTE_UPS.APPLICATION_SETUP.PERMISSIONS,
        parentKey: 'ups-application-setup',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_permission_list],
      },
    ],
  },
  {
    key: 'ups-organization-setup',
    label: 'Organization Setup',
    isTitle: false,
    icon: 'dripicons-network-3',
    permissions: [
      PERMISSIONS.ups_view_organization_list,
      PERMISSIONS.ups_view_site_list,
      PERMISSIONS.ups_view_locale_list,
    ],
    children: [
      {
        key: 'ups-organizations',
        label: 'Organizations',
        url: ROUTE_UPS.ORGANIZATION_SETUP.ORGANIZATIONS,
        parentKey: 'ups-organization-setup',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_organization_list],
      },
      {
        key: 'ups-sites',
        label: 'Sites',
        url: ROUTE_UPS.ORGANIZATION_SETUP.SITES,
        parentKey: 'ups-organization-setup',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_site_list],
      },
      {
        key: 'ups-locales',
        label: 'Locales',
        url: ROUTE_UPS.ORGANIZATION_SETUP.LOCALES,
        parentKey: 'ups-organization-setup',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_locale_list],
      },
    ],
  },
  {
    key: 'ups-user-management',
    label: 'User Management',
    isTitle: false,
    icon: 'dripicons-user-group',
    permissions: [
      PERMISSIONS.ups_view_user_list,
      PERMISSIONS.ups_view_user_permissions_list,
      PERMISSIONS.ups_view_roles_to_assign_list
    ],
    children: [
      {
        key: 'ups-users',
        label: 'Users',
        url: ROUTE_UPS.USER_MANAGEMENT.USERS,
        parentKey: 'ups-user-management',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_user_list],
      },
      {
        key: 'ups-user-permissions',
        label: 'User Permissions',
        url: ROUTE_UPS.USER_MANAGEMENT.USER_PERMISSIONS,
        parentKey: 'ups-user-management',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_user_list],
      },
      {
        key: 'ups-roles-to-assign',
        label: 'Roles to Assign',
        url: ROUTE_UPS.USER_MANAGEMENT.ROLES_TO_ASSIGN,
        parentKey: 'ups-user-management',
        isTitle: false,
        permissions: [PERMISSIONS.ups_view_roles_to_assign_list],
      },
    ],
  },
]
