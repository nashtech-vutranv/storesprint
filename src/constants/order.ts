export const orderStatus = {
    TRANSFERRING: 'Transferring',
    PROCESSING: 'Processing',
    TRANSFER_FAILED: 'Transfer failed',
    CANCELLED: 'Cancelled',
    CANCELLED_PART: 'Cancelled (Part)',
    DISPATCHED: 'Dispatched',
    DISPATCHED_PART: 'Dispatched (Part)',
    DISPATCHED_CANCEL: 'Dispatched (Canc)',
    CONFIRMED: 'Confirmed',
    CONFIRMED_DISPATCHED_PART: 'Confirmed Dispatched (Part)',
    CONFIRMED_DISPATCHED_CANCEL: 'Confirmed Dispatched (Canc)',
    CANCELLED_CONFIRMED: 'Cancelled Confirmed'
}