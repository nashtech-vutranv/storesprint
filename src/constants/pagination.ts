import {IPagination} from '../interface'

export const defaultTablePaginationSortByErpIdPerPage: IPagination = {
  first: 0,
  rows: 10,
  page: 1,
  sortField: 'erpId',
  sortOrder: 1,
  keyword: '',
}

export const defaultAllDatasByErpIdPagination: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  rows: 200000
}

export const defaultOrganizationsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage
}

export const defaultAggregatorsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'name'
}

export const defaultMarketplacesTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'name'
}

export const defaultRelationshipAMsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'aggregator.name'
}

export const defaultAssignedResourceTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'organizationResponse.name',
}

export const defaultRelationshipTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'organization.name'
}

export const defaultUsersTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'fullName'
}

export const defaulStockLevelTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'organizationWarehouse.name'
}

export const defaulProductPriceTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'siteLocale.site.name'
}

export const defaulMarketplaceTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'localeMarketplace.marketplace.name',
}

export const defaulOrderShipmentTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'shipment.shipmentNumber'
}

export const defaultListingStateTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'marketplaceProduct.product.erpId'
}

export const defaultMappingsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  rows: 3000,
  sortField: 'mappingType.name',
}

export const defaultPropertyValuesTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'organizationName',
}

export const productCategoryMappingsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'mmsValue',
}

export const productPropertyMappingsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'attributeResponse.value',
}

export const productPropertyValueMappingTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'mappingTarget.marketplaceValue',
}

export const courierMappingsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'mmsValue',
}

export const returnsRefundsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'modifiedAt',
}

export const deliveryTypeMappingsTablePaginationPerPage: IPagination = {
  ...defaultTablePaginationSortByErpIdPerPage,
  sortField: 'mmsValue',
}
