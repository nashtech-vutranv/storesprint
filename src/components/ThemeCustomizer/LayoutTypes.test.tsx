import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import * as LayoutConstants from '../../constants'
import LayoutTypes from './LayoutTypes'

test('renders LayoutTypes', async () => {
  const tree = render(
    <MemoryRouter>
      <LayoutTypes
        changeLayoutType={jest.fn()}
        layoutType='horizontal'
        layoutConstants={LayoutConstants.LayoutTypes}
      />
    </MemoryRouter>
  )

  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
})
