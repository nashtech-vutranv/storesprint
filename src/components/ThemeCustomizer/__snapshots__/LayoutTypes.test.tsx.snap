// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LayoutTypes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h5
        class="mt-3"
      >
        Layout
      </h5>
      <hr
        class="mt-1"
      />
      <div
        class="form-check form-switch mb-1 form-check"
      >
        <input
          class="form-check-input"
          id="vertical-layout"
          name="layout-type"
          type="radio"
          value="vertical"
        />
        <label
          class="form-check-label"
          for="vertical-layout"
        >
          Vertical Layout
        </label>
      </div>
      <div
        class="form-check form-switch mb-1 form-check"
      >
        <input
          class="form-check-input"
          id="horizontal-layout"
          name="layout-type"
          type="radio"
          value="topnav"
        />
        <label
          class="form-check-label"
          for="horizontal-layout"
        >
          Horizontal Layout
        </label>
      </div>
      <div
        class="form-check form-switch mb-1 form-check"
      >
        <input
          class="form-check-input"
          id="detached-layout"
          name="layout-type"
          type="radio"
          value="detached"
        />
        <label
          class="form-check-label"
          for="detached-layout"
        >
          Detached Layout
        </label>
      </div>
      <div
        class="form-check form-switch mb-1 form-check"
      >
        <input
          class="form-check-input"
          id="full-mode-layout"
          name="layout-type"
          type="radio"
          value="full"
        />
        <label
          class="form-check-label"
          for="full-mode-layout"
        >
          Full Layout
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <h5
      class="mt-3"
    >
      Layout
    </h5>
    <hr
      class="mt-1"
    />
    <div
      class="form-check form-switch mb-1 form-check"
    >
      <input
        class="form-check-input"
        id="vertical-layout"
        name="layout-type"
        type="radio"
        value="vertical"
      />
      <label
        class="form-check-label"
        for="vertical-layout"
      >
        Vertical Layout
      </label>
    </div>
    <div
      class="form-check form-switch mb-1 form-check"
    >
      <input
        class="form-check-input"
        id="horizontal-layout"
        name="layout-type"
        type="radio"
        value="topnav"
      />
      <label
        class="form-check-label"
        for="horizontal-layout"
      >
        Horizontal Layout
      </label>
    </div>
    <div
      class="form-check form-switch mb-1 form-check"
    >
      <input
        class="form-check-input"
        id="detached-layout"
        name="layout-type"
        type="radio"
        value="detached"
      />
      <label
        class="form-check-label"
        for="detached-layout"
      >
        Detached Layout
      </label>
    </div>
    <div
      class="form-check form-switch mb-1 form-check"
    >
      <input
        class="form-check-input"
        id="full-mode-layout"
        name="layout-type"
        type="radio"
        value="full"
      />
      <label
        class="form-check-label"
        for="full-mode-layout"
      >
        Full Layout
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
